src/main/java/com/minecolonies/coremod/colony/colonyEvents/raidEvents/babarianEvent/BarbarianRaidEvent.java

```diff
@@ -0,0 +1,426 @@
+        }
+
+        if (entity instanceof EntityArcherBarbarian)
+        {
+            archers.remove(entity);
+            horde.numberOfArchers--;
+        }
+
+        if (entity instanceof EntityBarbarian)
+        {
+            barbarians.remove(entity);
+            horde.numberOfRaiders--;
+        }
+
+        horde.hordeSize--;
+
+        if (horde.hordeSize == 0)
+        {
+            status = EventStatus.DONE;
+        }
+
+        sendHordeMessage();
+    }
+
+    @Override
+    public void onStart()
+    {
+        final BlockPos spawnPos = PirateEventUtils.getLoadedPositionTowardsCenter(spawnPoint, colony, MAX_SPAWN_DEVIATION, spawnPoint, MIN_CENTER_DISTANCE, 10);
+        if (spawnPos == null)
+        {
+            status = EventStatus.CANCELED;
+            return;
+        }
+
+        status = EventStatus.PROGRESSING;
+        RaiderMobUtils.spawn(BARBARIAN, horde.numberOfRaiders, spawnPos, colony.getWorld(), colony, id);
+        RaiderMobUtils.spawn(CHIEF, horde.numberOfBosses, spawnPos, colony.getWorld(), colony, id);
+        RaiderMobUtils.spawn(ARCHER, horde.numberOfArchers, spawnPos, colony.getWorld(), colony, id);
+
+        LanguageHandler.sendPlayersMessage(
+          colony.getImportantMessageEntityPlayers(),
+          RAID_EVENT_MESSAGE + horde.getMessageID(), colony.getName(), BlockPosUtil.calcDirection(colony.getCenter(), spawnPoint));
+    }
+
+    @Override
+    public void onUpdate()
+    {
+        if (horde.hordeSize == 0)
+        {
+            status = EventStatus.DONE;
+        }
+
+        if (!respawns.isEmpty())
+        {
+            for (final Tuple<ResourceLocation, BlockPos> entry : respawns)
+            {
+                final BlockPos spawnPos = PirateEventUtils.getLoadedPositionTowardsCenter(entry.getSecond(), colony, MAX_RESPAWN_DEVIATION, spawnPoint, MIN_CENTER_DISTANCE, 10);
+                if (spawnPos != null)
+                {
+                    RaiderMobUtils.spawn(entry.getFirst(), 1, spawnPos, colony.getWorld(), colony, id);
+                }
+            }
+            respawns.clear();
+            return;
+        }
+
+        if (chiefs.size() + archers.size() + barbarians.size() < horde.numberOfBosses + horde.numberOfRaiders + horde.numberOfArchers)
+        {
+            final BlockPos spawnPos = PirateEventUtils.getLoadedPositionTowardsCenter(spawnPoint, colony, MAX_RESPAWN_DEVIATION, spawnPoint, MIN_CENTER_DISTANCE, 10);
+            if (spawnPos != null)
+            {
+                RaiderMobUtils.spawn(CHIEF, horde.numberOfBosses - chiefs.size(), spawnPos, colony.getWorld(), colony, id);
+                RaiderMobUtils.spawn(ARCHER, horde.numberOfArchers - archers.size(), spawnPos, colony.getWorld(), colony, id);
+                RaiderMobUtils.spawn(BARBARIAN, horde.numberOfRaiders - barbarians.size(), spawnPos, colony.getWorld(), colony, id);
+            }
+        }
+
+
+            for (final Entity entity : getEntities())
```

Comment:
formatting here too

Reply:
oh and the debug logs are still in, those arent needed